src/common/exporter/scrivener/v2/utils.js

```diff
@@ -0,0 +1,171 @@
+    .join(', ')
+}
+
+// When a character is created from a template, it gets all of the template
+// attributes attached to its object. This method turns any template attribute
+// attached to a character into a single object where the key is the attribute
+// name and the value is the value the user entered for that attribute
+export function buildTemplateProperties(templates) {
+  return templates.reduce((acc, { attributes }) => {
+    attributes.forEach((attribute) => (acc[attribute.name] = attribute.value))
+    return acc
+  }, {})
+}
+
+// id starts at 2 because the base_scrivx.json already uses ids
+// 0, 1, and 2 and the first thing the createBinderItem functions do
+// is increment the id
+let nextId = 2
+export function resetId() {
+  nextId = 2
+}
+
+// convenience method for creating a folder BinderItem
+export function createFolderBinderItem(title) {
+  nextId++
+  let data = cloneDeep(binderItem)
+  data['_attributes']['ID'] = nextId
+  data['_attributes']['Type'] = 'Folder'
+  data['Title']['_text'] = title
+  data['Children'] = { BinderItem: [] }
+  return {
+    id: nextId,
+    binderItem: data,
+  }
+}
+
+// convenience method for creating a text binder item
+export function createTextBinderItem(text) {
+  nextId++
+  let data = cloneDeep(binderItem)
+  data['_attributes']['ID'] = nextId
+  data['Title']['_text'] = text
+  return {
+    id: nextId,
+    binderItem: data,
+  }
+}
+
+// characters, places, and notes all have a number of fields that a user can enter
+// to describe the object. We want to include all of those fields in the RTF document
+// that we create for the object. This function takes an object of key value pairs
+// and generates a rich text document where the keys are headings and the values are
+// paragraphs that can then be used to create a RTF document
+export function buildDescriptionFromObject(object) {
+  const description = []
+  for (const [key, value] of Object.entries(object)) {
+    // if the user hasn't input a value we won't include it
+    // for rich content the check is abit more involved
+    if (isPropertyEmpty(value)) {
+      continue
+    }
+
+    description.push({
+      type: 'heading-two',
+      children: [
+        {
+          text: key,
+        },
+      ],
+    })
+
+    if (typeof value === 'string') {
+      description.push({
+        type: 'paragraph',
+        children: [{ text: value }],
+      })
+    } else {
+      description.push(...value)
+    }
+  }
+  return description
+}
+
+// this function checks if a value from building a description is empty
+// RichContent has a few gotchas around what is considered "empty"
+export function isPropertyEmpty(property) {
+  if (!property) return true
+
+  // Rich content is an array
+  const isRichContent = Array.isArray(property)
+
+  // If it isn't rich content and it's not empty then it has a value
+  if (!isRichContent) return false
+
+  // if it's empty it only has one item and the type is blank
```

Comment:
this one might not actually be accurate. This is the default/initial value for rich content:

`[{ type: 'paragraph', children: [{ text: '' }] }]`

Before, the initial value didn't have a `type: 'paragraph'` so I wonder if the file you're working with is like that maybe?

Reply:
Ok I will add this as well.